### PR TITLE
Added templating capability for the API Docs

### PR DESCRIPTION
**Pull request by dibgerge**
*Issue #, if available:*

*Description of changes:*

Added custom Jinja templates for automatic generation of api docs. Modified the toc table titles to remove the annoying "package" word in every title.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -269,6 +270,6 @@ def setup(app):
 linkcheck_retries = 3
 linkcheck_workers = int(multiprocessing.cpu_count() / 2)
 
-intersphinx_mapping = {
-    "mxnet": ("https://mxnet.apache.org/api/python/docs", None)
-}
+# intersphinx_mapping = {
+#     "mxnet": ("https://mxnet.apache.org/api/python/docs", None)
+# }
```
Comment (jaheba) [inline docs/conf.py]: Let's remove these lines.

File: src/gluonts/model/deepar/_network.py [hunk 2]
```diff
@@ -406,7 +406,7 @@ def hybrid_forward(
             dim=1,
         )
 
-        # mask the loss at one time step iff one or more observations is missing in the target dimensions
+        # mask the loss at one time step iff one or more observations is missing in  the target dimensions
```
Comment (jaheba) [inline src/gluonts/model/deepar/_network.py]: This shouldn't be part of this PR.

File: docs/Makefile [hunk 3]
```diff
@@ -10,7 +10,7 @@ BUILDDIR      = _build
 
 
 APIDOC        = sphinx-apidoc
-APIDOC_OPTS   = --implicit-namespaces --separate --module-first
+APIDOC_OPTS   = --implicit-namespaces --separate --module-first -t apidoc
```
Comment (jaheba) [inline docs/Makefile]:
```suggestion
APIDOC_OPTS   = --implicit-namespaces --separate --module-first --templatedir apidoc
```

We also need to update the sphinx version to support this.

File: docs/Makefile [hunk 4]
```diff
@@ -23,7 +23,6 @@ apidoc:
 	@rm -Rf api/$(APIDOC_ROOT)
 	@$(APIDOC) $(APIDOC_OPTS) -o api/$(APIDOC_ROOT) ../src/$(APIDOC_ROOT) setup* test docs *pycache*
 	@rm -Rf api/$(APIDOC_ROOT)/modules.rst
-	@sed -i"" -e "s/$(APIDOC_ROOT) package/API Docs/" api/$(APIDOC_ROOT)/$(APIDOC_ROOT).rst
```
Comment (jaheba) [inline docs/Makefile]: Can you explain why you removed this line?

File: docs/apidoc/toc.rst_t [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{{ header | heading }}
+
+.. toctree::
+   :maxdepth: {{ maxdepth }}
+{% for docname in docnames %}
+   {{ docname }}
+{%- endfor %}
+
```
Comment (jaheba) [inline docs/apidoc/toc.rst_t]: Isn't this the same file as the original one?

File: docs/conf.py [hunk 6]
```diff
@@ -195,10 +195,11 @@
         "https://docs.python.org/{.major}".format(sys.version_info),
         None,
     ),
-    "mxnet": ("https://mxnet.apache.org/", None),
+    # "mxnet": ("https://mxnet.apache.org/", None),
```
Comment (lostella) [inline docs/conf.py]: Also this one can just be deleted I guess
